rust/tool/src/install.rs

```diff
@@ -332,7 +335,8 @@ impl Installer {
     fn install_systemd_boot(&self) -> Result<()> {
         let systemd_boot = self
             .systemd
-            .join("lib/systemd/boot/efi/systemd-bootx64.efi");
+            .join("lib/systemd/boot/efi")
+            .join(self.target_arch.systemd_filename());
```

Comment:
This is where it shows that information is leaking when `Architecture` is in `esp.rs`. This path has nothing to do with the esp.

Reply:
Completely agreed.

Reply:
But then, are you okay with the ext trait solution? Should I move it into systemd.rs ?